net/netcheck/netcheck.go

```diff
@@ -1476,11 +1476,33 @@ func (c *Client) addReportHistoryAndSetPreferredDERP(r *Report) {
 		}
 	}
 
+	// Scale each region's best latency by any provided weights from the
+	// DERPMap, for use in comparison below.
+	var weights map[int]float64
+	if dm.Hints != nil {
+		weights = dm.Hints.RegionWeight
+	}
+	for regionID, d := range bestRecent {
+		if weight := weights[regionID]; weight != 0 {
```

Comment:
`weight != 0` or `ok` ? Your call. But an explicit map entry with a value of 0 should be have its behavior documented, whatever it is.

Reply:
IMO, weights should not be 0 (or negative); I'll document this and update it to check `weight > 0` to ignore such weights.